imap/jmap_api.c

```diff
@@ -3093,6 +3195,14 @@ HIDDEN int jmap_set_sharewith(struct mailbox *mbox,
         }
     }
 
+    if (!r) {
+        /* Set proper access rights on JMAP upload folder */
+        /* XXX This is currently done by brute force.
```

Comment:
This comment might be better placed at the start of set_upload_rights. Without knowing the internals of that function, it is not clear where the "brute force" is coming from here.